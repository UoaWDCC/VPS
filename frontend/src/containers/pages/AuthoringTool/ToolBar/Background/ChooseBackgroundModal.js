import React from "react";
import ReactDOM from "react-dom";
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import MuiDialogTitle from "@material-ui/core/DialogTitle";
import MuiDialogContent from "@material-ui/core/DialogContent";
import MuiDialogActions from "@material-ui/core/DialogActions";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
import Typography from "@material-ui/core/Typography";

const modalRoot = document.querySelector("#modal-root");

const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(2),
  },
  closeButton: {
    position: "absolute",
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
});

const DialogTitle = withStyles(styles)((props) => {
  const { children, classes, onClose, ...other } = props;
  return (
    <MuiDialogTitle disableTypography className={classes.root} {...other}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButton
          aria-label="close"
          className={classes.closeButton}
          onClick={onClose}
        >
          <CloseIcon />
        </IconButton>
      ) : null}
    </MuiDialogTitle>
  );
});

export default function ChooseBackgroundModal({ isShowing, hide }) {
  console.log(isShowing);
  return ReactDOM.createPortal(
    <div>
      <Dialog
        onClose={hide}
        aria-labelledby="customized-dialog-title"
        open={isShowing}
      >
        <DialogTitle id="customized-dialog-title" onClose={hide}>
          Modal title
        </DialogTitle>
        <MuiDialogContent dividers>
          <Typography gutterBottom>
            Cras mattis consectetur purus sit amet fermentum. Cras justo odio,
            dapibus ac facilisis in, egestas eget quam. Morbi leo risus, porta
            ac consectetur ac, vestibulum at eros.
          </Typography>
          <Typography gutterBottom>
            Praesent commodo cursus magna, vel scelerisque nisl consectetur et.
            Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor
            auctor.
          </Typography>
          <Typography gutterBottom>
            Aenean lacinia bibendum nulla sed consectetur. Praesent commodo
            cursus magna, vel scelerisque nisl consectetur et. Donec sed odio
            dui. Donec ullamcorper nulla non metus auctor fringilla.
          </Typography>
        </MuiDialogContent>
        <MuiDialogActions>
          <Button autoFocus onClick={hide} color="primary">
            Save changes
          </Button>
        </MuiDialogActions>
      </Dialog>
    </div>,
    modalRoot
  );
}
