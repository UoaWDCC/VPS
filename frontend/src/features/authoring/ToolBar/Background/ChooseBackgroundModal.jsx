import Button from "@material-ui/core/Button";
import { useContext, useState } from "react";
import ImageListContainer from "components/ListContainer/ImageListContainer";
import ModalDialogue from "components/ModalDialogue";
import SceneContext from "context/SceneContext";
import ToolbarContext from "context/ToolbarContext";
import { useGet } from "hooks/crudHooks";
import styles from "./ChooseBackgroundModal.module.scss";
import { addImage } from "../ToolBarActions";

/**
 * This component is a modal that displays the default images for users to select.
 * @component
 */
export default function ChooseBackgroundModal({ isShowing, hide }) {
  const { currentScene, setCurrentScene } = useContext(SceneContext);
  const [images, setImages] = useState();
  const { handleDropdownClose } = useContext(ToolbarContext);

  const [selectedImage, setSelectedImage] = useState();
  useGet("/api/image", setImages);

  function closeDialogues() {
    hide();
    handleDropdownClose();
  }

  function handleButtonClick() {
    closeDialogues();
    addImage(currentScene, setCurrentScene, selectedImage);
  }

  const saveButton = (
    <Button
      autoFocus
      onClick={handleButtonClick}
      className={styles.dialogueActionButton}
      disabled={!selectedImage}
    >
      Add image
    </Button>
  );

  return (
    <ModalDialogue
      title="Choose Image"
      isShowing={isShowing}
      hide={closeDialogues}
      dialogueAction={saveButton}
    >
      <ImageListContainer data={images} onItemSelected={setSelectedImage} />
    </ModalDialogue>
  );
}
