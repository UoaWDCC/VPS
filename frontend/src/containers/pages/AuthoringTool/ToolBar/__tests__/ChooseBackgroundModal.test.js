import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import ChooseBackgroundModal from "../Background/ChooseBackgroundModal";

test("Scenario Selection page snapshot test", () => {
  const component = renderer.create(
    <BrowserRouter>
      <ChooseBackgroundModal />
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
