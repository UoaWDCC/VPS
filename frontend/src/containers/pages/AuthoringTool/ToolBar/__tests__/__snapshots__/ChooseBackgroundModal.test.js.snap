// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseBackgroundModal component snapshot test 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div />
  </div>
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="customized-dialog-title"
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root Component-root-1"
          id="customized-dialog-title"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Choose Image
          </h6>
          <button
            aria-label="close"
            class="MuiButtonBase-root MuiIconButton-root Component-closeButton-2"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiDialogContent-root MuiDialogContent-dividers"
        >
          <ul
            class="MuiImageList-root"
            style="margin: -5px;"
          />
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text dialogueActionButton Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add image
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
